File: source/SharpFlame/frmWZLoad.cs


using System;


namespace SharpFlame
{
    public partial class frmWZLoad
    {
        public clsOutput Output;
        public string[] lstMap_MapName;

        public frmWZLoad(string[] MapNames, clsOutput NewOutput, string FormTitle)
        {
            InitializeComponent();

            Icon = App.ProgramIcon;

            Output = NewOutput;
            Output.Result = -1;

            var A = 0;

            lstMap.Items.Clear();
            lstMap_MapName = MapNames;
            for ( A = 0; A <= MapNames.GetUpperBound(0); A++ )
            {
                lstMap.Items.Add(MapNames[A]);
            }

            Text = FormTitle;
        }

        public void lstMaps_DoubleClick(object sender, EventArgs e)
        {
            if ( lstMap.SelectedIndex >= 0 )
            {
                Output.Result = lstMap.SelectedIndex;
                Close();
            }
        }

        public class clsOutput
        {
            public int Result;
        }

        public struct sMapNameList
        {
            public string[] Names;
        }
    }
}

File: reference/FlaMEVB/frmWZLoad.vb

Public Class frmWZLoad

    Public Class clsOutput
        Public Result As Integer
    End Class

    Public lstMap_MapName() As String

    Public Output As clsOutput

    Public Structure sMapNameList
        Public Names() As String
    End Structure

    Public Sub New(MapNames() As String, NewOutput As clsOutput, FormTitle As String)
        InitializeComponent()

        Icon = ProgramIcon

        Output = NewOutput
        Output.Result = -1

        Dim A As Integer

        lstMap.Items.Clear()
        lstMap_MapName = MapNames
        For A = 0 To MapNames.GetUpperBound(0)
            lstMap.Items.Add(MapNames(A))
        Next

        Text = FormTitle
    End Sub

    Private Sub lstMaps_DoubleClick(sender As Object, e As System.EventArgs) Handles lstMap.DoubleClick

        If lstMap.SelectedIndex >= 0 Then
            Output.Result = lstMap.SelectedIndex
            Close()
        End If
    End Sub
End Class
